src/pricer.py

```diff
@@ -591,29 +589,34 @@ def __init__(self, contract: EuropeanBarrierContract, model: MarketModel, params
         super().__init__(contract, model, params)
 
         if isinstance(model, FlatVolModel):
-            self._mc_method = MCMethodFlatVol(self._contract, self._model, self._params)
+            self.mc_method = MCMethodFlatVol(self.contract, self.model, self.params)
+        elif isinstance(model, BSVolModel):
+            self.mc_method = MCMethodBS(self.contract, self.model, self.params)
         else:
             raise TypeError(f'MC is not supported for model type {type(contract).__name__}')
 
-    def calc_fair_value(self) -> float:
-            contract = self._contract
-            # num_mon_mod = round(contract._num_mon / 10)    # BB specific
-            num_mon_mod = contract._num_mon
+    def calc_fair_value_with_ci(self) -> float:
+            contract = self.contract
+            # num_mon_mod = round(contract.num_mon / 10)    # BB specific
```

Comment:
Could you please remove commented code?

Reply:
deleted (however, would make sense to always scale down the nr of monitoring points for the BB pricer...)